lib/hanami/utils/inflector.rb

```diff
@@ -27,17 +27,31 @@ def add(key, value)
         # @since 0.6.0
         # @api private
         def ===(other)
-          key = other.downcase
+          key = if other.downcase =~ /_(\w*)\z/
+            $1
+          else
+            other.downcase
+          end
           @rules.key?(key) || @rules.value?(key)
         end
 
         # @since 0.6.0
         # @api private
         def apply(string)
-          key    = string.downcase
+          key = if string.downcase =~ /_(\w*)\z/
```

Comment:
Please consider to extract a private method for this logic, as it's exactly the same used for `#===`